storage/redis/redis_scheduler_event_storage_test.go

```diff
@@ -9,109 +9,111 @@
 package redis
 
 import (
-	"encoding/json"
-	"errors"
 	"fmt"
 	"testing"
 	"time"
 
-	"github.com/go-redis/redis"
-	"github.com/golang/mock/gomock"
 	uuid "github.com/satori/go.uuid"
-	redismocks "github.com/topfreegames/extensions/redis/mocks"
-	"github.com/topfreegames/maestro/models"
 	. "github.com/topfreegames/maestro/models"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-)
 
-func TestModels(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "Storage Suite")
-}
+	"github.com/topfreegames/extensions/redis"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+
+	"github.com/topfreegames/maestro/extensions"
+	mtesting "github.com/topfreegames/maestro/testing"
+)
 
 var (
 	schedulerName   string
-	mockCtrl        *gomock.Controller
-	mockRedisClient *redismocks.MockRedisClient
-	mockPipeline    *redismocks.MockPipeliner
+	redisClient *redis.Client
+	logger      *logrus.Logger
+	hook        *test.Hook
+	eventStorage *RedisSchedulerEventStorage
 )
 
+func TestIntStorage(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Models Integration Suite")
+}
+
+var _ = BeforeSuite(func() {
+	config, err := mtesting.GetDefaultConfig()
+	Expect(err).NotTo(HaveOccurred())
+
+	logger, hook = test.NewNullLogger()
+	logger.Level = logrus.DebugLevel
+
+	redisClient, err = extensions.GetRedisClient(logger, config)
+	Expect(err).NotTo(HaveOccurred())
+
+	eventStorage = NewRedisSchedulerEventStorage(redisClient.Client)
+})
+
 var _ = Describe("Scheduler events", func() {
 	BeforeEach(func() {
 		schedulerName = uuid.NewV4().String()
-		mockCtrl = gomock.NewController(GinkgoT())
-		mockRedisClient = redismocks.NewMockRedisClient(mockCtrl)
-		mockPipeline = redismocks.NewMockPipeliner(mockCtrl)
-	})
-
-	AfterEach(func() {
-		mockCtrl.Finish()
 	})
 
 	Describe("PersistSchedulerEvent", func() {
 		It("should return no error when event is persisted successfully", func() {
 			metadata := make(map[string]interface{})
 			metadata["reason"] = "rollback"
 
-			mockRedisClient.EXPECT().TxPipeline().Return(mockPipeline)
-			mockPipeline.EXPECT().ZAdd(fmt.Sprintf("scheduler:%s:events", schedulerName), gomock.Any()).Times(1)
-			mockPipeline.EXPECT().ZRemRangeByScore(fmt.Sprintf("scheduler:%s:events", schedulerName), "-inf", gomock.Any()).Times(1)
-			mockPipeline.EXPECT().Exec().Times(1)
-
 			event := NewSchedulerEvent("UPDATE_STARTED", schedulerName, metadata)
 
-			err := NewRedisSchedulerEventStorage(mockRedisClient).PersistSchedulerEvent(event)
+			err := eventStorage.PersistSchedulerEvent(event)
+			Expect(err).To(BeNil())
+
+			events, err := eventStorage.LoadSchedulerEvents(schedulerName, 1)
 			Expect(err).To(BeNil())
+			Expect(events).To(HaveLen(1))
 		})
 
-		It("should return error when redis pipe execution throws error", func() {
+		It("should not crop events older than limit", func() {
```

Comment:
Not sure if the title matches what the test is doing since it is removing the `oldEvent`.